src/Command/CommandBase.php

```diff
@@ -885,6 +885,12 @@ protected function selectRemoteContainer(InputInterface $input, $includeWorkers
         }
 
         $environment = $this->getSelectedEnvironment();
+
+        // Enterprise environments do not have seperate containers for workers.
+        if ($includeWorkers && $environment->deployment_target === 'enterprise') {
```

Comment:
Unfortunately the deployment_target for an Enterprise (dedicated) environment isn't necessarily the string `enterprise`. Elsewhere we check for `!== 'local'`, which is hacky but works.

Reply:
For this reason - the slightly unreliable hack - I want to hide the interactive worker choice for non-`local` environments, but still respect the `--worker` option.

Reply:
@pjcdawkins updated, I haven't tested it however.